pkg/upgrade/cache.go

```diff
@@ -0,0 +1,264 @@
+package upgrade
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/0-fs/meta"
+	"github.com/threefoldtech/0-fs/storage"
+	"github.com/xxtea/xxtea-go/xxtea"
+	"golang.org/x/crypto/blake2b"
+
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// DefaultDownloadWorkers define the default number of workload to use to downloads data blocks
+	DefaultDownloadWorkers = 4
+	//DefaultBlockSize is the default block size
+	DefaultBlockSize = 512 //KB
```

Comment:
Please add unit suffix e.g DefaultBlockSizeKB

Reply:
also noticed these constants are also used internally in functions and there's no way to override them, please make them unexported